mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -1559,6 +1571,90 @@ vectorizeAsTensorPackOp(RewriterBase &rewriter, tensor::PackOp packOp,
   return success();
 }
 
+/// Vectorize a `tensor::UnPackOp` without OuterDimsPerms to these 4 Ops:
+///   Vector::TransferReadOp - Reads a vector from the source tensor
+///   vector::TransposeOp - Transpose the Source tensor
+///   ShapeCastOp - Reshape the data based on the target.
+///   vector::TransferWriteOp. - Write the result vector back to the destination
+///   tensor
+static LogicalResult vectorizeAsUnpackOp(RewriterBase &rewriter,
+                                         tensor::UnPackOp unpackOp,
+                                         ArrayRef<int64_t> inputVectorSizes,
+                                         SmallVectorImpl<Value> &newResults) {
+
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPoint(unpackOp);
+
+  RankedTensorType unpackTensorType = unpackOp.getSourceType();
+
+  SmallVector<int64_t> readMaskShape(unpackTensorType.getShape());
```

Comment:
I think we should try to support outer_dims_perm cases here. I believe it should work as long as the readMaskShape is corrected for outer_dims_perm here. The following should work:

1. initialize `readMaskShape` equal to the `inputVectorSizes`
2. Divide by inner_tiles the same as before
3. Apply outer_dims_perm to `readMaskShape`
4. Append the inner_tiles to the end of `readMaskShape`

Reply:
this [c7ed75e](https://github.com/llvm/llvm-project/pull/76087/commits/c7ed75e39f79fdf8c4de880c7ea8d1800be347d4) should support outer_dims_perm, however I think your steps 2 and 3 should be switched because in the manual it says outer_dims_perm should be done before combining any other dimensions.